### ISSUE TITLE
Indentation rule change in 0.40.0

### ISSUE THREAD

**Issue by jai** [pinterest/ktlint#1012]
cc @umbertosidoti @theerasan

Hi - it seems that this is new in 0.40.0, and I think it may be related to:
* https://github.com/pinterest/ktlint/issues/921
* https://github.com/pinterest/ktlint/pull/950
* https://github.com/pinterest/ktlint/issues/938

Any info or guidance would be appreciated. Have considered creating our own ruleset but wanted to understand why this is happening first.

## Expected Behavior
Indentation `indent` error shouldn't be triggered (`ktlint` should pass) for these lines:

```kotlin
class ApplyEmailViewModel @Inject constructor(
    viewModelExecutor: ViewModelExecutor,
    private val navigator: MainNavigator,
    private val interactor: EmailInteractor,
    private val viewStateMapper: EmailToEmailViewStateMapper,
    private val analytics: ApplyEmailAnalytics
) : BaseViewModelEventLiveData<
    ApplyEmailViewState,       # Line 22
    ApplyEmailEvent.              # Line 23
    >(viewModelExecutor),    # Line 24
    FormFieldListener {
```

## Observed Behavior
`ktlint` fails with errors:

```bash
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:22:1: Unexpected indentation (4) (should be 8) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:23:1: Unexpected indentation (4) (should be 8) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:24:1: Unexpected indentation (4) (should be 8) (indent)
```

## Steps to Reproduce
Here's the code:
```kotlin
class ApplyEmailViewModel @Inject constructor(
    viewModelExecutor: ViewModelExecutor,
    private val navigator: MainNavigator,
    private val interactor: EmailInteractor,
    private val viewStateMapper: EmailToEmailViewStateMapper,
    private val analytics: ApplyEmailAnalytics
) : BaseViewModelEventLiveData<
    ApplyEmailViewState,
    ApplyEmailEvent
    >(viewModelExecutor),
    FormFieldListener {

    init {
        compositeDisposable.add(
            interactor.emailListenable.listen(false) {
                onEmailUpdated(it)
            }
        )
    }
}
```

## Your Environment

* Version of ktlint used: 0.40.0 (vs 0.39.0)
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): just plain old `ktlint` and `ktlint --verbose`
* Version of Gradle used (if applicable): N/A
* Operating System and version: macOS Big Sur
* Link to your project (if it's a public repository): N/A

**Comment by jai** [comment]
Here's conflicting output between ktlint 0.39.0 and 0.40.0 (the first run is from 0.40.0):

```bash
(⎈ |test-api:default)❯ ktlint --verbose
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:22:1: Unexpected indentation (4) (should be 8) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:23:1: Unexpected indentation (4) (should be 8) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:24:1: Unexpected indentation (4) (should be 8) (indent)

~/Developer/android-app card_info*
(⎈ |test-api:default)❯ ./ktlint --version
0.39.0

~/Developer/android-app ci-add-custom-ktlint*
(⎈ |test-api:default)❯ ./ktlint --verbose
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:22:1: Unexpected indentation (8) (should be 4) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:23:1: Unexpected indentation (8) (should be 4) (indent)
/Users/jai/Developer/android-app/presentation/apply-presentation/src/main/kotlin/com/honestbank/android/apply/presentation/email/ApplyEmailViewModel.kt:24:1: Unexpected indentation (8) (should be 4) (indent)
```

**Comment by t-kameyama** [comment]
simple code to reproduce:

```kotlin
open class Foo<A, B>(s: String)

interface Bar

class Test(
    private val s: String
) : Foo<
        String,
        Int
        >(s),
    Bar
```

**Comment by t-kameyama** [comment]
It seems this issue has been resolved by https://github.com/pinterest/ktlint/pull/1006

**Comment by romtsn** [comment]
@jai I wasn't able to reproduce this on master, could you give it a try with the latest snapshot?
